Objects/genobject.c

```diff
@@ -1338,9 +1336,9 @@ compute_cr_origin(int origin_depth, _PyInterpreterFrame *current_frame)
     frame = current_frame;
     for (int i = 0; i < frame_count; ++i) {
         PyCodeObject *code = frame->f_code;
-        PyObject *frameinfo = Py_BuildValue("OiO",
-                                            code->co_filename,
-                                            PyCode_Addr2Line(frame->f_code, frame->f_lasti*sizeof(_Py_CODEUNIT)),
+        int addr = _PyInterpreterFrame_LASTI(frame) * sizeof(_Py_CODEUNIT);
+        int line = PyCode_Addr2Line(frame->f_code, addr);
```

Comment:
Slightly off-topic, but I wonder why we are going to the expense of computing the line here, rather than just storing the instruction offset?